### PR TITLE
tools: add interrupt support to host exerciser memory

### PR DESCRIPTION
**Pull request by anandaravuri**
- add new command line input parameter interrupt
- register host exerciser memory interrupt and triggered by  afu after coemption of test.

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 1]
```diff
@@ -111,6 +111,11 @@ class host_exerciser_cmd : public test_command
               he_lpbk_cfg_.TputInterleave = host_exe_->he_interleave_;
         }
 
+        // Set Interrupt test mode
+        if (host_exe_->he_interrupt_ > 0) {
```
Comment (tswhison) [inline samples/host_exerciser/host_exerciser_cmd.h:115]: The parameter is declared to accept a Range(0, 3).  Doesn't this mean that 0 is a valid interrupt index?

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 2]
```diff
@@ -212,6 +234,7 @@ class host_exerciser_cmd : public test_command
     shared_buffer::ptr_t source_;
     shared_buffer::ptr_t destination_;
     shared_buffer::ptr_t dsm_;
+    he_interrupt0 he_intrrupt_;
```
Comment (tswhison) [inline samples/host_exerciser/host_exerciser_cmd.h:237]: s/he_intrrupt_/he_interrupt_/

Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h]: fixed

File: samples/host_exerciser/host_exerciser_cmd.h (hunk 1 above)
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h:115]: interrupt index has to fixed during integration. we don't know interrupt number

**Review by tswhison — APPROVED** [review]

**Review by tswhison — APPROVED** [review]

**Pull request merged**